### ISSUE TITLE
ConTeXt pdf-engine fails on KOMAoptions

### ISSUE THREAD

**Issue by cswingle** [quarto-dev/quarto-cli#412]
When trying to produce a PDF via ConTeXt I get an "undefined control sequence" error associated with the line: `\KOMAoptions{captions}{tableheading}`.

I don't really know what KOMAoptions are, but a google search suggests it's associated with a LaTeX macro package. Maybe this `header-includes` should only be used when LaTeX is the `pdf-engine`?

```bash
$ quarto --version
0.9.86
$ context --version
mtx-context     | ConTeXt Process Management 1.04
mtx-context     |
mtx-context     | main context file: ${HOME}/context/tex/texmf-context/tex/context/base/mkiv/context.mkiv
mtx-context     | current version: 2022.01.21 20:13
mtx-context     | main context file: ${HOME}/context/tex/texmf-context/tex/context/base/mkxl/context.mkxl
mtx-context     | current version: 2022.01.21 20:13
```

Example qmd:

```markdown
---
title: "Test"
format:
  pdf:
    pdf-engine: context
---

# Introduction

Rendering this document yields an error on line 63, which is 
`\KOMAoptions{captions}{tableheading}`
```

Render:

```bash
$ quarto render README.qmd --to pdf
pandoc 
  to: pdf
  output-file: README.pdf
  standalone: true
  pdf-engine: context
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  filters:
    - crossref
  
metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  title: Test
  
Error producing PDF.
tex error       > tex error on line 63 in file ./input.tex: Undefined control sequence \undefined 
```

**Comment by jjallaire** [comment]
Thanks for reporting this! Should be fixed here https://github.com/quarto-dev/quarto-cli/commit/6a6f9e88721c207455e2984a0d742d6cee5a4be3 (fix available in binary version v0.9.87: https://github.com/quarto-dev/quarto-cli/releases/latest)

**Comment by cswingle** [comment]
@jjallaire I just installed the 0.9.87 release and now it doesn't give an error when generating the PDF, but the PDF isn't valid:

```bash
$ quarto render README.qmd --to pdf
pandoc 
  to: pdf
  output-file: README.pdf
  standalone: true
  pdf-engine: context
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  filters:
    - crossref
  
metadata
  title: Test

$ pdfinfo README.pdf
Syntax Error (11780): Illegal character '}'
Syntax Error: Couldn't find trailer dictionary
Syntax Error (11780): Illegal character '}'
Syntax Error: Couldn't find trailer dictionary
Syntax Error: Couldn't read xref table
```

I'm not sure what to make of that. I tried including `keep-tex: true`, but I don't get a `README.tex` file. When I dump to ConTeXt instead of PDF, the `KOMAOptions` are in the output, and the header info looks a little strange (`pdf-engine: xelatex` ?):

```bash
$ quarto render README.qmd --to context
pandoc 
  to: context
  output-file: README.tex
  standalone: true
  pdf-engine: xelatex
  variables:
    graphics: true
    tables: true
  default-image-extension: pdf
  filters:
    - crossref
  
metadata
  documentclass: scrartcl
  classoption:
    - DIV=11
    - numbers=noendperiod
  papersize: letter
  header-includes:
    - '\KOMAoption{captions}{tableheading}'
  title: Test
  
Output created: README.tex
$ context README.tex
...
tex error       > tex error on line 63 in file ./README.tex: Undefined control sequence \undefined 

<line 3.63> 
    \KOMAoption
    {captions}{tableheading} 
...
```

**Comment by jjallaire** [comment]
I think there must still be crosstalk between the standard pdf format and context that we need to untangle. I just got context installed locally and will spend time tomorrow making sure that we get all of this right. Thanks again for reporting this and will hopefully fully resolve soon!

**Comment by jjallaire** [comment]
Okay, as suspected there were multiple things wrong with our handling of context. Should all be resolved here: https://github.com/quarto-dev/quarto-cli/commit/3f149c8b72f20ce751c67c7a6ffa82e30d084d00 (available in v0.9.90 here: https://github.com/quarto-dev/quarto-cli/releases/latest).

**Comment by jjallaire** [comment]
Okay, great to hear!
